### PR TITLE
feat: code editor details

### PR DESCRIPTION
**Pull request by shouhardik**
### Description
- The code editor has been made using Regex without the help of any external library.

### Related Issues
- https://github.com/cse210-fa24-group14/cse210-fa24-group14/issues/64.

### Type of Change
- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [x] Refactor
- [ ] Other (please specify):

### Testing Details
- Tested each of the feature.

### Checklist
- [x] Code follows the style guide.
- [x] I have performed a self-review of my code.
- [ ] Documentation has been updated (if necessary).
- [ ] Tests have been added or updated.

### Sample
- image

### CODE REVIEW

**Review by kenny-lfx — APPROVED** [review]

**Comment by philemonputra** [comment]
There seems to be an offset to the curser and where I am typing. Please try to fix. 
Also, there seems to be a lag when I type in the code section. 

image

**Comment by rashi-bhansali** [comment]
@sho

> There seems to be an offset to the curser and where I am typing. Please try to fix. Also, there seems to be a lag when I type in the code section.
> 
> image

@shouhardik can we reduce the debounce time for highlighted code to show up on the syntax overlay?

**Comment by shouhardik** [comment]
> @sho
> 
> > There seems to be an offset to the curser and where I am typing. Please try to fix. Also, there seems to be a lag when I type in the code section.
> > image
> 
> @shouhardik can we reduce the debounce time for highlighted code to show up on the syntax overlay?

The Debounce time is already low to 300 ms.

File: src/js/components/NotesView.js [hunk 1]
```diff
@@ -8,6 +9,7 @@ export class NotesView {
     this.onDeleteCell = null;
     this.onAddCell = null;
     this.onUpdateCell = null;
+    //this.syntaxHighlight = new SyntaxHighlighter()
```
Comment (philemonputra) [inline src/js/components/NotesView.js:12]: Delete this comment.

Comment (shouhardik) [reply src/js/components/NotesView.js]: Done @philemonputra

**Pull request merged**